go-licenses PATH is incorrect

We get this error if $GOPATH/bin is not on the path 
```
Collecting licenses
hack/generate-statik.sh: line 40: go-licenses: command not found
Makefile:228: recipe for target 'generate-statik' failed
make: *** [generate-statik] Error 127
```

Also placing the go-licenses binary in GOPATH/bin is inconsistent with the statik binary placement (and other tool binaries) it should live in `hack/bin`. 

This is currently blocking the release, marking as P0.

--- a/hack/generate-statik.sh
+++ b/hack/generate-statik.sh
@@ -20,7 +20,7 @@ DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
 
 BIN=${DIR}/bin
 STATIK=${BIN}/statik
-LICENSES=go-licenses
+LICENSES=${BIN}/go-licenses
 
 TMP_DIR=$(mktemp -d)
 trap "rm -rf $TMP_DIR" EXIT
@@ -31,7 +31,7 @@ if ! [ -x "$(command -v ${LICENSES})" ]; then
     # from a dependency.
     echo "Installing go-licenses"
     pushd $(mktemp -d)
-    go mod init tmp; go get -mod='' github.com/google/go-licenses
+    go mod init tmp; GOBIN=${BIN} go get -mod='' github.com/google/go-licenses
     popd
 fi
 

